crates/sncast/src/helpers/fee.rs

```diff
@@ -151,3 +151,38 @@ pub enum FeeSettings {
         max_gas_unit_price: Option<u128>,
     },
 }
+
+#[macro_export]
+macro_rules! impl_from_version_for_fee_token {
```

Comment:
I really like the solution, although i wonder: 
Couldn't this be solved with a trait? 
Something like `PayableTransaction`, which would expose the mapping of version -> token 
And also would define a method for formatting the error

WDYT?

Reply:
I like this idea but I want to be sure we're on the same page:

I am thinking to have a trait like this:

```rust
trait PayableTransaction {
    fn error_message(&self, token: &str, version: &str) -> String;
    fn validate(&self) -> Result<()>;
    fn version_from_token(&self) -> Option<FeeToken>;
}
```

and implement it via a macro:

```rust
impl_payable_transaction!(Invoke, token_not_supported_for_invoke,
    InvokeVersion::V1 => FeeToken::Eth,
    InvokeVersion::V3 => FeeToken::Strk
);
```

Reply:
Yep i am not against implementing it via a macro
Having the clear interface without `cargo expand` is what i'm after